handler/oauth2/flow_revoke.go

```diff
@@ -0,0 +1,37 @@
+package oauth2
+
+import (
+	"github.com/ory-am/fosite"
+	"golang.org/x/net/context"
+)
+
+type TokenRevocationHandler struct {
+	TokenRevocationStorage TokenRevocationStorage
+}
+
+// RevokeToken implements https://tools.ietf.org/html/rfc7009#section-2.1
+// The token type hint indicates which token type check should be performed first.
+func (r *TokenRevocationHandler) RevokeToken(ctx context.Context, token string, tokenType fosite.TokenType) error {
+	revokeFuncs := []func(){
+		func() {
+			// Refresh token
+			signature := r.TokenRevocationStorage.RefreshTokenSignature(token)
+			r.TokenRevocationStorage.RevokeRefreshToken(ctx, signature)
```

Comment:
Any suggestions?

Should there be an additional Storage function such as:

```DiscoverToken(ctx, signature, tokenType) (<something>, error)```

```go
func() {
			// Refresh token
			signature := r.TokenRevocationStorage.RefreshTokenSignature(token)
			r.TokenRevocationStorage.RevokeRefreshToken(ctx, signature)
			if something, err := r.TokenRevocationStorage.DiscoverToken(ctx, signature, tokenType); err == nil {
				r.DoThing()
			}
		},
```

Or could `Revoke[Access|Refresh]Token` accept an additional signature and the discovery takes place beforehand?